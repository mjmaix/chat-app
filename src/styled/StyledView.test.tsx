import React from 'react';
import { render } from 'react-native-testing-library';
import 'jest-styled-components/native';

import { StyledView } from './StyledView';
import { Text } from 'react-native';

const createTestProps = (props?: object) => ({
  testID: 'testid',
  ...props,
});

test('it works', () => {
  const props = createTestProps();
  const { getByTestId } = render(
    <StyledView {...props}>
      <Text>TEST</Text>
    </StyledView>,
  );
  expect(getByTestId('testid')).toMatchSnapshot();
});
