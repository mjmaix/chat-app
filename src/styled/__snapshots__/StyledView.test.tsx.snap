// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
ReactTestInstance {
  "_fiber": FiberNode {
    "_debugHookTypes": null,
    "_debugID": 4,
    "_debugIsCurrentlyTiming": false,
    "_debugOwner": null,
    "_debugSource": null,
    "alternate": null,
    "child": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 5,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": [Circular],
      "_debugSource": null,
      "alternate": null,
      "child": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 6,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": [Circular],
        "_debugSource": null,
        "alternate": null,
        "child": FiberNode {
          "_debugHookTypes": null,
          "_debugID": 7,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": [Circular],
          "_debugSource": null,
          "alternate": null,
          "child": FiberNode {
            "_debugHookTypes": null,
            "_debugID": 8,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": [Circular],
            "_debugSource": null,
            "alternate": null,
            "child": FiberNode {
              "_debugHookTypes": null,
              "_debugID": 9,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": null,
              "_debugSource": null,
              "alternate": null,
              "child": FiberNode {
                "_debugHookTypes": null,
                "_debugID": 10,
                "_debugIsCurrentlyTiming": false,
                "_debugOwner": [Circular],
                "_debugSource": null,
                "alternate": null,
                "child": FiberNode {
                  "_debugHookTypes": null,
                  "_debugID": 11,
                  "_debugIsCurrentlyTiming": false,
                  "_debugOwner": null,
                  "_debugSource": null,
                  "alternate": null,
                  "child": null,
                  "childExpirationTime": 0,
                  "contextDependencies": null,
                  "effectTag": 0,
                  "elementType": null,
                  "expirationTime": 0,
                  "firstEffect": null,
                  "index": 0,
                  "key": null,
                  "lastEffect": null,
                  "memoizedProps": "TEST",
                  "memoizedState": null,
                  "mode": 0,
                  "nextEffect": null,
                  "pendingProps": "TEST",
                  "ref": null,
                  "return": [Circular],
                  "sibling": null,
                  "stateNode": Object {
                    "isHidden": false,
                    "tag": "TEXT",
                    "text": "TEST",
                  },
                  "tag": 6,
                  "type": null,
                  "updateQueue": null,
                },
                "childExpirationTime": 0,
                "contextDependencies": null,
                "effectTag": 0,
                "elementType": "Text",
                "expirationTime": 0,
                "firstEffect": null,
                "index": 0,
                "key": null,
                "lastEffect": null,
                "memoizedProps": Object {
                  "children": "TEST",
                },
                "memoizedState": null,
                "mode": 0,
                "nextEffect": null,
                "pendingProps": Object {
                  "children": "TEST",
                },
                "ref": null,
                "return": [Circular],
                "sibling": null,
                "stateNode": Object {
                  "children": Array [
                    Object {
                      "isHidden": false,
                      "tag": "TEXT",
                      "text": "TEST",
                    },
                  ],
                  "isHidden": false,
                  "props": Object {
                    "children": "TEST",
                  },
                  "rootContainerInstance": Object {
                    "children": Array [
                      Object {
                        "children": Array [
                          [Circular],
                        ],
                        "isHidden": false,
                        "props": Object {
                          "children": <Text>
                            TEST
                          </Text>,
                          "style": Array [
                            Object {},
                          ],
                          "testID": "testid",
                        },
                        "rootContainerInstance": [Circular],
                        "tag": "INSTANCE",
                        "type": "View",
                      },
                    ],
                    "createNodeMock": [Function],
                    "tag": "CONTAINER",
                  },
                  "tag": "INSTANCE",
                  "type": "Text",
                },
                "tag": 5,
                "type": "Text",
                "updateQueue": null,
              },
              "childExpirationTime": 0,
              "contextDependencies": null,
              "effectTag": 1,
              "elementType": [Function],
              "expirationTime": 0,
              "firstEffect": null,
              "index": 0,
              "key": null,
              "lastEffect": null,
              "memoizedProps": Object {
                "children": "TEST",
              },
              "memoizedState": null,
              "mode": 0,
              "nextEffect": null,
              "pendingProps": Object {
                "children": "TEST",
              },
              "ref": null,
              "return": [Circular],
              "sibling": null,
              "stateNode": Component {
                "_reactInternalFiber": [Circular],
                "_reactInternalInstance": Object {},
                "context": Object {},
                "props": Object {
                  "children": "TEST",
                },
                "refs": Object {},
                "state": null,
                "updater": Object {
                  "enqueueForceUpdate": [Function],
                  "enqueueReplaceState": [Function],
                  "enqueueSetState": [Function],
                  "isMounted": [Function],
                },
              },
              "tag": 1,
              "type": [Function],
              "updateQueue": null,
            },
            "childExpirationTime": 0,
            "contextDependencies": null,
            "effectTag": 0,
            "elementType": "View",
            "expirationTime": 0,
            "firstEffect": null,
            "index": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": Object {
              "children": <Text>
                TEST
              </Text>,
              "style": Array [
                Object {},
              ],
              "testID": "testid",
            },
            "memoizedState": null,
            "mode": 0,
            "nextEffect": null,
            "pendingProps": Object {
              "children": <Text>
                TEST
              </Text>,
              "style": Array [
                Object {},
              ],
              "testID": "testid",
            },
            "ref": null,
            "return": [Circular],
            "sibling": null,
            "stateNode": Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "isHidden": false,
                      "tag": "TEXT",
                      "text": "TEST",
                    },
                  ],
                  "isHidden": false,
                  "props": Object {
                    "children": "TEST",
                  },
                  "rootContainerInstance": Object {
                    "children": Array [
                      [Circular],
                    ],
                    "createNodeMock": [Function],
                    "tag": "CONTAINER",
                  },
                  "tag": "INSTANCE",
                  "type": "Text",
                },
              ],
              "isHidden": false,
              "props": Object {
                "children": <Text>
                  TEST
                </Text>,
                "style": Array [
                  Object {},
                ],
                "testID": "testid",
              },
              "rootContainerInstance": Object {
                "children": Array [
                  [Circular],
                ],
                "createNodeMock": [Function],
                "tag": "CONTAINER",
              },
              "tag": "INSTANCE",
              "type": "View",
            },
            "tag": 5,
            "type": "View",
            "updateQueue": null,
          },
          "childExpirationTime": 0,
          "contextDependencies": null,
          "effectTag": 1,
          "elementType": [Function],
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "children": <Text>
              TEST
            </Text>,
            "style": Array [
              Object {},
            ],
            "testID": "testid",
          },
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "children": <Text>
              TEST
            </Text>,
            "style": Array [
              Object {},
            ],
            "testID": "testid",
          },
          "ref": null,
          "return": [Circular],
          "sibling": null,
          "stateNode": Component {
            "_reactInternalFiber": [Circular],
            "_reactInternalInstance": Object {},
            "context": Object {},
            "props": Object {
              "children": <Text>
                TEST
              </Text>,
              "style": Array [
                Object {},
              ],
              "testID": "testid",
            },
            "refs": Object {},
            "state": null,
            "updater": Object {
              "enqueueForceUpdate": [Function],
              "enqueueReplaceState": [Function],
              "enqueueSetState": [Function],
              "isMounted": [Function],
            },
          },
          "tag": 1,
          "type": [Function],
          "updateQueue": null,
        },
        "childExpirationTime": 0,
        "contextDependencies": Object {
          "expirationTime": 0,
          "first": Object {
            "context": Object {
              "$$typeof": Symbol(react.context),
              "Consumer": Object {
                "$$typeof": Symbol(react.context),
                "_calculateChangedBits": null,
                "_context": [Circular],
              },
              "Provider": Object {
                "$$typeof": Symbol(react.provider),
                "_context": [Circular],
              },
              "_calculateChangedBits": null,
              "_currentRenderer": null,
              "_currentRenderer2": null,
              "_currentValue": undefined,
              "_currentValue2": undefined,
              "_threadCount": 0,
            },
            "next": null,
            "observedBits": 1073741823,
          },
        },
        "effectTag": 1,
        "elementType": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": Object {
            "$$typeof": Symbol(react.context),
            "Consumer": [Circular],
            "Provider": Object {
              "$$typeof": Symbol(react.provider),
              "_context": [Circular],
            },
            "_calculateChangedBits": null,
            "_currentRenderer": null,
            "_currentRenderer2": null,
            "_currentValue": undefined,
            "_currentValue2": undefined,
            "_threadCount": 0,
          },
        },
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {
          "children": [Function],
        },
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": Object {
          "children": [Function],
        },
        "ref": null,
        "return": [Circular],
        "sibling": null,
        "stateNode": null,
        "tag": 9,
        "type": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": Object {
            "$$typeof": Symbol(react.context),
            "Consumer": [Circular],
            "Provider": Object {
              "$$typeof": Symbol(react.provider),
              "_context": [Circular],
            },
            "_calculateChangedBits": null,
            "_currentRenderer": null,
            "_currentRenderer2": null,
            "_currentValue": undefined,
            "_currentValue2": undefined,
            "_threadCount": 0,
          },
        },
        "updateQueue": null,
      },
      "childExpirationTime": 0,
      "contextDependencies": null,
      "effectTag": 1,
      "elementType": [Function],
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": <Text>
          TEST
        </Text>,
        "forwardedComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "displayName": "Styled(View)",
          "inlineStyle": InlineStyle {
            "rules": Array [],
          },
          "render": [Function],
          "styledComponentId": "StyledNativeComponent",
          "target": [Function],
          "withComponent": [Function],
        },
        "forwardedRef": null,
        "testID": "testid",
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "children": <Text>
          TEST
        </Text>,
        "forwardedComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "displayName": "Styled(View)",
          "inlineStyle": InlineStyle {
            "rules": Array [],
          },
          "render": [Function],
          "styledComponentId": "StyledNativeComponent",
          "target": [Function],
          "withComponent": [Function],
        },
        "forwardedRef": null,
        "testID": "testid",
      },
      "ref": null,
      "return": [Circular],
      "sibling": null,
      "stateNode": StyledNativeComponent {
        "_reactInternalFiber": [Circular],
        "_reactInternalInstance": Object {},
        "attrs": Object {},
        "context": Object {},
        "props": Object {
          "children": <Text>
            TEST
          </Text>,
          "forwardedComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "displayName": "Styled(View)",
            "inlineStyle": InlineStyle {
              "rules": Array [],
            },
            "render": [Function],
            "styledComponentId": "StyledNativeComponent",
            "target": [Function],
            "withComponent": [Function],
          },
          "forwardedRef": null,
          "testID": "testid",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
        },
        "warnAttrsFnObjectKeyDeprecated": [Function],
        "warnInnerRef": [Function],
        "warnNonStyledComponentAttrsObjectKey": [Function],
      },
      "tag": 1,
      "type": [Function],
      "updateQueue": null,
    },
    "childExpirationTime": 0,
    "contextDependencies": null,
    "effectTag": 1,
    "elementType": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "displayName": "Styled(View)",
      "inlineStyle": InlineStyle {
        "rules": Array [],
      },
      "render": [Function],
      "styledComponentId": "StyledNativeComponent",
      "target": [Function],
      "withComponent": [Function],
    },
    "expirationTime": 0,
    "firstEffect": null,
    "index": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {
      "children": <Text>
        TEST
      </Text>,
      "testID": "testid",
    },
    "memoizedState": null,
    "mode": 0,
    "nextEffect": null,
    "pendingProps": Object {
      "children": <Text>
        TEST
      </Text>,
      "testID": "testid",
    },
    "ref": null,
    "return": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 2,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": null,
      "_debugSource": null,
      "alternate": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 2,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": null,
        "_debugSource": null,
        "alternate": [Circular],
        "child": null,
        "childExpirationTime": 0,
        "contextDependencies": null,
        "effectTag": 0,
        "elementType": null,
        "expirationTime": 1073741823,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "sibling": null,
        "stateNode": Object {
          "containerInfo": Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "children": Array [
                      Object {
                        "isHidden": false,
                        "tag": "TEXT",
                        "text": "TEST",
                      },
                    ],
                    "isHidden": false,
                    "props": Object {
                      "children": "TEST",
                    },
                    "rootContainerInstance": [Circular],
                    "tag": "INSTANCE",
                    "type": "Text",
                  },
                ],
                "isHidden": false,
                "props": Object {
                  "children": <Text>
                    TEST
                  </Text>,
                  "style": Array [
                    Object {},
                  ],
                  "testID": "testid",
                },
                "rootContainerInstance": [Circular],
                "tag": "INSTANCE",
                "type": "View",
              },
            ],
            "createNodeMock": [Function],
            "tag": "CONTAINER",
          },
          "context": Object {},
          "current": [Circular],
          "didError": false,
          "earliestPendingTime": 0,
          "earliestSuspendedTime": 0,
          "expirationTime": 0,
          "finishedWork": null,
          "firstBatch": null,
          "hydrate": false,
          "latestPendingTime": 0,
          "latestPingedTime": 0,
          "latestSuspendedTime": 0,
          "nextExpirationTimeToWorkOn": 0,
          "nextScheduledRoot": null,
          "pendingChildren": null,
          "pendingCommitExpirationTime": 0,
          "pendingContext": null,
          "pingCache": null,
          "timeoutHandle": -1,
        },
        "tag": 3,
        "type": null,
        "updateQueue": Object {
          "baseState": null,
          "firstCapturedEffect": null,
          "firstCapturedUpdate": null,
          "firstEffect": null,
          "firstUpdate": Object {
            "callback": null,
            "expirationTime": 1073741823,
            "next": null,
            "nextEffect": null,
            "payload": Object {
              "element": <ForwardRef
                testID="testid"
              >
                <Text>
                  TEST
                </Text>
              </ForwardRef>,
            },
            "tag": 0,
          },
          "lastCapturedEffect": null,
          "lastCapturedUpdate": null,
          "lastEffect": null,
          "lastUpdate": Object {
            "callback": null,
            "expirationTime": 1073741823,
            "next": null,
            "nextEffect": null,
            "payload": Object {
              "element": <ForwardRef
                testID="testid"
              >
                <Text>
                  TEST
                </Text>
              </ForwardRef>,
            },
            "tag": 0,
          },
        },
      },
      "child": [Circular],
      "childExpirationTime": 0,
      "contextDependencies": null,
      "effectTag": 0,
      "elementType": null,
      "expirationTime": 0,
      "firstEffect": [Circular],
      "index": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": null,
      "memoizedState": Object {
        "element": <ForwardRef
          testID="testid"
        >
          <Text>
            TEST
          </Text>
        </ForwardRef>,
      },
      "mode": 0,
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "sibling": null,
      "stateNode": Object {
        "containerInfo": Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "isHidden": false,
                      "tag": "TEXT",
                      "text": "TEST",
                    },
                  ],
                  "isHidden": false,
                  "props": Object {
                    "children": "TEST",
                  },
                  "rootContainerInstance": [Circular],
                  "tag": "INSTANCE",
                  "type": "Text",
                },
              ],
              "isHidden": false,
              "props": Object {
                "children": <Text>
                  TEST
                </Text>,
                "style": Array [
                  Object {},
                ],
                "testID": "testid",
              },
              "rootContainerInstance": [Circular],
              "tag": "INSTANCE",
              "type": "View",
            },
          ],
          "createNodeMock": [Function],
          "tag": "CONTAINER",
        },
        "context": Object {},
        "current": [Circular],
        "didError": false,
        "earliestPendingTime": 0,
        "earliestSuspendedTime": 0,
        "expirationTime": 0,
        "finishedWork": null,
        "firstBatch": null,
        "hydrate": false,
        "latestPendingTime": 0,
        "latestPingedTime": 0,
        "latestSuspendedTime": 0,
        "nextExpirationTimeToWorkOn": 0,
        "nextScheduledRoot": null,
        "pendingChildren": null,
        "pendingCommitExpirationTime": 0,
        "pendingContext": null,
        "pingCache": null,
        "timeoutHandle": -1,
      },
      "tag": 3,
      "type": null,
      "updateQueue": Object {
        "baseState": Object {
          "element": <ForwardRef
            testID="testid"
          >
            <Text>
              TEST
            </Text>
          </ForwardRef>,
        },
        "firstCapturedEffect": null,
        "firstCapturedUpdate": null,
        "firstEffect": null,
        "firstUpdate": null,
        "lastCapturedEffect": null,
        "lastCapturedUpdate": null,
        "lastEffect": null,
        "lastUpdate": null,
      },
    },
    "sibling": null,
    "stateNode": null,
    "tag": 11,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "displayName": "Styled(View)",
      "inlineStyle": InlineStyle {
        "rules": Array [],
      },
      "render": [Function],
      "styledComponentId": "StyledNativeComponent",
      "target": [Function],
      "withComponent": [Function],
    },
    "updateQueue": null,
  },
}
`;
